cmd/aks-periscope/aks-periscope.go

```diff
@@ -32,40 +30,166 @@ func main() {
+			osmCollector.GetName():             osmCollector,
+		})
+
+	//diagnosers
+	//NOTE currently the collector instance objects are shared between the collector itself and things which use it as a dependency
+	networkConfigDiagnoser := diagnoser.NewNetworkConfigDiagnoser(dnsCollector, kubeletCmdCollector, selectedExporters)
+	networkOutboundDiagnoser := diagnoser.NewNetworkOutboundDiagnoser(networkOutboundCollector, selectedExporters)
+	selectedDiagnosers := selectDiagnosers(
+		map[string]interfaces.Diagnoser{
+			networkConfigDiagnoser.GetName():   networkConfigDiagnoser,
+			networkOutboundDiagnoser.GetName(): networkOutboundDiagnoser,
+		})
+
+	return selectedCollectors, selectedDiagnosers, selectedExporters
+}
+
+// selectCollectors select the collectors to run
+func selectCollectors(allCollectorsByName map[string]interfaces.Collector) []interfaces.Collector {
 	collectors := []interfaces.Collector{}
-	containerLogsCollector := collector.NewContainerLogsCollector(exporter)
-	networkOutboundCollector := collector.NewNetworkOutboundCollector(5, exporter)
-	dnsCollector := collector.NewDNSCollector(exporter)
-	kubeObjectsCollector := collector.NewKubeObjectsCollector(exporter)
-	systemLogsCollector := collector.NewSystemLogsCollector(exporter)
-	ipTablesCollector := collector.NewIPTablesCollector(exporter)
-	nodeLogsCollector := collector.NewNodeLogsCollector(exporter)
-	kubeletCmdCollector := collector.NewKubeletCmdCollector(exporter)
-	systemPerfCollector := collector.NewSystemPerfCollector(exporter)
-	helmCollector := collector.NewHelmCollector(exporter)
-	osmCollector := collector.NewOsmCollector(exporter)
-
-	collectors = append(collectors, containerLogsCollector)
-	collectors = append(collectors, dnsCollector)
-	collectors = append(collectors, kubeObjectsCollector)
-	collectors = append(collectors, networkOutboundCollector)
 
-	if contains(collectorList, "connectedCluster") {
-		collectors = append(collectors, helmCollector)
+	//read list of collectors that are enabled
+	var enabledCollectorNames []string
+
+	//TODO try get partners to move from COLLECTOR_LIST to use ENABLED_COLLECTORS instead, for now COLLECTOR_LIST takes precedence if defined
+	collectorList := strings.Fields(os.Getenv("COLLECTOR_LIST"))
+	if collectorList != nil {
+		enabledCollectorNames = selectCollectorsUsingCollectorList(collectorList)
 	} else {
-		collectors = append(collectors, systemLogsCollector)
-		collectors = append(collectors, ipTablesCollector)
-		collectors = append(collectors, nodeLogsCollector)
-		collectors = append(collectors, kubeletCmdCollector)
-		collectors = append(collectors, systemPerfCollector)
+		enabledCollectorNames = strings.Fields(os.Getenv("ENABLED_COLLECTORS"))
 	}
 
+	enabledCollectorNames = strings.Fields(os.Getenv("ENABLED_COLLECTORS"))
+
+	for _, collectorName := range enabledCollectorNames {
+		collectors = append(collectors, allCollectorsByName[collectorName])
+	}
+
+	return collectors
+}
```

Comment:
introduce new ENABLED_COLLECTORS as intended replacement for COLLECTOR_LIST. Support COLLECTOR_LIST for back-compat